cgo/logic.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2021 Matrix Origin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mo_impl.h"
+
+// Logical and operator
+int32_t Logic_VecAnd(void *r, void *a, void  *b, uint64_t n, uint64_t *anulls, uint64_t *bnulls, uint64_t *rnulls, int32_t flag) {
+    bool *rt = (bool *) r;
+    bool *at = (bool *) a;
+    bool *bt = (bool *) b;
+    if ((flag & LEFT_IS_SCALAR) != 0) {
+        for (uint64_t i = 0; i < n; i++) {
+            rt[i] = at[0] && bt[i];
+        }
+        if (rnulls != NULL) {
+            if (!at[0]) {
+                for (uint64_t i = 0; i < n; i++) {
+                    if (Bitmap_Contains(rnulls, i)) {
```

Comment:
Isn't this just clear all bitmap?

Reply:
Yes, in this case, all bitmap elements will be reset to 0